Dockerfile

```
@@ -0,0 +1,18 @@
+FROM docker.io/library/rust:alpine AS builder
+# RUN wget -O source.tar.gz $(wget -qO- https://api.github.com/repos/mbrubeck/agate/releases/latest | sed -nE 's/^.*"tarball_url"\s*:\s*"([^"]+)".*$/\1/p')&& tar xzf source.tar.gz && \
+#     mv mbrubeck-agate-* /agate
+
+WORKDIR /agate
+COPY src src
+COPY Cargo.toml .
+COPY Cargo.lock .
+COPY Cross.toml .
+RUN apk --no-cache add libc-dev && \
+    cargo install --target x86_64-unknown-linux-musl --path .
```

Given that `apk add` may take some time to complete, I think it'd be smart to move this before the `COPY` commands, so the `apk` step doesn't need to be repeated if the code changes. (If the package repo changes more frequently than the code then the `COPY` steps are significantly faster)

This does mean _two_ image layers, but here that's a feature, as one will be dependent on external software (`apk`), and the other dependent on this repo's code.

(I agree with your choice not to include an `apk update --no-cache` call here, for stability; as any important security updates will be addressed by the very broadly pinned `FROM docker.io/library/rust:alpine`)

```suggestion
RUN apk --no-cache add libc-dev

COPY src src
COPY Cargo.toml .
COPY Cargo.lock .
COPY Cross.toml .
RUN cargo install --target x86_64-unknown-linux-musl --path .
```